src/ray/gcs/entry_change_notification.h

```diff
@@ -0,0 +1,51 @@
+#ifndef RAY_GCS_ENTRY_CHANGE_NOTIFICATION_H
+#define RAY_GCS_ENTRY_CHANGE_NOTIFICATION_H
+
+#include <ray/protobuf/gcs.pb.h>
+#include <vector>
+
+namespace ray {
+
+namespace gcs {
+
+/// \class EntryChangeNotification
+/// EntryChangeNotification class is a template class which represent
+/// notification of entry change from GCS.
+template <typename Data>
+class EntryChangeNotification {
+ public:
+  EntryChangeNotification(rpc::GcsChangeMode change_mode, std::vector<Data> data)
+      : change_mode_(change_mode), data_(std::move(data)) {}
+
+  EntryChangeNotification(EntryChangeNotification &&other) {
+    change_mode_ = other.change_mode_;
+    data_ = std::move(other.data_);
+  }
+
+  EntryChangeNotification &operator=(EntryChangeNotification &&other) {
+    change_mode_ = other.change_mode_;
+    data_ = std::move(other.data_);
+  }
```

Comment:
Consider disallow copy and assign, and instead passing around unique_ptr<EntryChangeNotification>.

Reply:
`EntryChangeNotification` is exposed to the user as the `second  parameter` of `SubscribeCallback`：

`template <typename ID, typename Data>
using SubscribeCallback = std::function<void(const ID &id, const Data &result)>;`

So unique_ptr it's not suitable here. 

We used `move construction and move assignment` here to avoid copy `EntryChangeNotification`. The copy may happen when we construct `EntryChangeNotification` for the first time.
And I think the copy here cost little. But if you really wish to avoid copy data, maybe we can pass `const & data`
 when we construct `EntryChangeNotification`.